langchain4j/src/test/java/dev/langchain4j/service/AiServicesWithToolsIT.java

```diff
@@ -727,4 +720,119 @@ private static Map<String, Object> toMap(String arguments) {
+                "First call getTransactionAmount for T001, then for T002. Do not answer before you know all amounts!";
+
+        Result<AiMessage> response = assistant.chat(userMessage);
+
+        ToolExecutionRequest firstToolExecutionRequest = response.toolExecutions().get(0).toolExecutionRequest();
+        assertThat(firstToolExecutionRequest.name()).isEqualTo("getTransactionAmount");
+        assertThat(firstToolExecutionRequest.arguments())
+                .isEqualToIgnoringWhitespace("{\"arg0\": \"T001\"}");
+
+        ToolExecutionRequest secondToolExecutionRequest = response.toolExecutions().get(1).toolExecutionRequest();
+        assertThat(secondToolExecutionRequest.name()).isEqualTo("getTransactionAmount");
+        assertThat(secondToolExecutionRequest.arguments())
+                .isEqualToIgnoringWhitespace("{\"arg0\": \"T002\"}");
+
+        String firstResult = response.toolExecutions().get(0).toolExecutionResult();
+        assertThat(firstResult).isEqualTo("11.1");
+
+        String secondResult = response.toolExecutions().get(1).toolExecutionResult();
+        assertThat(secondResult).contains("22.2");
+
+    }
+
+    @ParameterizedTest
+    @MethodSource("modelsWithoutParallelToolCalling")
+    public void should_execute_multiple_tools_sequentially_and_context_included_in_result(ChatLanguageModel chatLanguageModel) {
+
+        TransactionService transactionService = spy(new TransactionService());
+
+        ChatMemory chatMemory = MessageWindowChatMemory.withMaxMessages(10);
+
+        ChatLanguageModel spyChatLanguageModel = spy(chatLanguageModel);
+
+        AssistantReturningResult assistant = AiServices.builder(AssistantReturningResult.class)
+                .chatLanguageModel(spyChatLanguageModel)
+                .chatMemory(chatMemory)
+                .tools(transactionService)
+                .build();
+
+        String userMessage = "What are the amounts of transactions T001 and T002?";
+
+        Result<AiMessage> response = assistant.chat(userMessage);
+
+        ToolExecutionRequest firstToolExecutionRequest = response.toolExecutions().get(0).toolExecutionRequest();
+        assertThat(firstToolExecutionRequest.name()).isEqualTo("getTransactionAmount");
+        assertThat(firstToolExecutionRequest.arguments())
+                .isEqualToIgnoringWhitespace("{\"arg0\": \"T001\"}");
+
+        ToolExecutionRequest secondToolExecutionRequest = response.toolExecutions().get(1).toolExecutionRequest();
+        assertThat(secondToolExecutionRequest.name()).isEqualTo("getTransactionAmount");
+        assertThat(secondToolExecutionRequest.arguments())
+                .isEqualToIgnoringWhitespace("{\"arg0\": \"T002\"}");
+
+        String firstResult = response.toolExecutions().get(0).toolExecutionResult();
+        assertThat(firstResult).isEqualTo("11.1");
+
+        String secondResult = response.toolExecutions().get(1).toolExecutionResult();
+        assertThat(secondResult).contains("22.2");
+
```

Comment:
please add an assertion to check that `spyChatLanguageModel` has been invoked 2 times